validphys2/src/validphys/loader.py

```diff
@@ -362,14 +363,40 @@ def check_fktable(self, theoryID, setname, cfac, use_fixed_predictions=False):
             fixed_predictions_path = theopath/ 'simu_factors' / ('SIMU_%s.yaml' % setname)
             cfactors = self.check_cfactor(theoryID, setname, cfac)
             return FKTableSpec(fkpath, cfactors, use_fixed_predictions=True, fixed_predictions_path=fixed_predictions_path)
+        
+        # use different file name for the FK table if the commondata is new
+        if new_commondata:
+            fkpath = tuple([theopath/ 'fastkernel' / (f'{setname}.pineappl.lz4')])
```

Comment:
This does not work for double differential measurements, i.e. `ATLAS_1JET_13TEV_DIF` which is a double differential measurement in |y| and pT, to make it work one could extract the file names from the `<DATASET>_<OBS>_metadata.yaml` file. I suggest moving the loading of the `.yaml` file from the next `if` clause to this one.

Reply:
In my case the dataset name is `ATLAS_1JET_13TEV_DIF_PT-Y`  but the names of the FK files are `ATLAS_1JET_13TEV_DIF_PT-Y_BIN#.pineappl.lz4`